pkg/cmd/controller/interfaces.go

```diff
@@ -107,21 +107,19 @@ func NewControllerContext(
 	openshiftControllerContext := &ControllerContext{
 		OpenshiftControllerConfig: config,
 
+		// k8s 1.21 rebase - SAControllerClientBuilder replaced with NewDynamicClientBuilder
+		// See https://github.com/kubernetes/kubernetes/pull/99291
 		ClientBuilder: OpenshiftControllerClientBuilder{
-			ControllerClientBuilder: clientbuilder.SAControllerClientBuilder{
-				ClientConfig:         rest.AnonymousClientConfig(clientConfig),
-				CoreClient:           kubeClient.CoreV1(),
-				AuthenticationClient: kubeClient.AuthenticationV1(),
-				Namespace:            defaultOpenShiftInfraNamespace,
-			},
+			ControllerClientBuilder: clientbuilder.NewDynamicClientBuilder(
+				rest.AnonymousClientConfig(clientConfig),
+				kubeClient.CoreV1(),
+				defaultOpenShiftInfraNamespace),
```

Comment:
/cc @sttts @deads2k 

Followed the lead on the upstream pull request (mentioned in the comment)

Reply:
Note - openshift-controller-manager needs permission to create serivceaccount tokens in the openshift-infra namespace to use the dynamic client builder. To be granted in https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/213